src/main/java/org/myrobotlab/service/Solr.java

```diff
@@ -983,52 +1118,67 @@ public void onMessage(Message message) {
   public static void main(String[] args) {
     LoggingFactory.init(Level.INFO);
     try {
+      Python python = (Python) Runtime.start("python", "Python");
       Solr solr = (Solr) Runtime.start("solr", "Solr");
       solr.startEmbedded();
-      // WebGui webgui = (WebGui)Runtime.start("webgui", "WebGui");
+      WebGui webgui = (WebGui)Runtime.start("webgui", "WebGui");
+      OpenCV cv = (OpenCV)Runtime.start("cv", "OpenCV");
+
+      Runtime.start("programab", "ProgramAB");
+      Runtime.start("clock", "Clock");
+      Runtime.start("ard", "Arduino");
+      
+      solr.attachAllInboxes();
+      solr.attachAllOutboxes();
+      // lets start indexing frames!
+      solr.attach(cv);
+      // and, go!
+      //cv.capture();
+
       // Create a test document
-      SolrInputDocument doc = new SolrInputDocument();
-      /*
-       * doc.setField("id", "Doc1"); doc.setField("title", "My title");
-       * doc.setField("content",
-       * "This is the text field, for a sample document in myrobotlab.  "); //
-       * add the document to the index solr.addDocument(doc); // commit the
-       * index solr.commit();
-       */
-
-      doc = new SolrInputDocument();
-      doc.setField("id", "Doc3");
-      doc.setField("title", "My title 3");
-      doc.setField("content", "This is the text field, for a sample document in myrobotlab. 2 ");
-      doc.setField("annoyance", 1);
-      // add the document to the index
-      solr.addDocument(doc);
-      // commit the index
-      solr.commit();
-
-      // search for the word myrobotlab
-      String queryString = "content:myrobotlab";
-      QueryResponse resp = solr.search(queryString);
-      for (int i = 0; i < resp.getResults().size(); i++) {
-        System.out.println("---------------------------------");
-        System.out.println("-- Printing Result number :" + i);
-        // grab a document out of the result set.
-        SolrDocument d = resp.getResults().get(i);
-        // iterate over the fields on the returned document
-        for (String fieldName : d.getFieldNames()) {
-
-          System.out.print(fieldName + "\t");
-          // fields can be multi-valued
-          for (Object value : d.getFieldValues(fieldName)) {
-            System.out.print(value);
-            System.out.print("\t");
+      boolean testIndex = false;
+      if (testIndex) {

```

Comment:
should be in unit test ... no ?

Reply:
there is a SolrTest ....  this is just some example code here.